api-platform/core
# More query parameter validation (follow #1692)

| Q             | A
| ------------- | ---
| Bug fix?      | kind of
| New feature?  | yes
| BC breaks?    | if filter config was wrongly configured
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1627 
| License       | MIT
| Doc PR        | Doc PR will follow if PR is approved

This follows #1692 

It implements all other validation filters from [OpenApi specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameter-object)

This might break wrongly configured api though. I see two possible solutions here:

  * make this a `[Might break]` as we can consider wrongly configured sites a "bug"
  * add a new configuration key `validate_filters`, default to `false`, and deprecate it being to `false`, encouraging people to set it to `true`, and remove the configuration key in 3.0
